golang/cosmos/x/swingset/client/cli/tx.go

```diff
@@ -117,6 +121,44 @@ func GetCmdProvisionOne() *cobra.Command {
 	return cmd
 }
 
+// GetCmdWalletAction is the CLI command for sending a WalletAction or WalletSpendAction transaction
+func GetCmdWalletAction() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "wallet-action [json string]",
+		Short: "perform a wallet action",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			owner := clientCtx.GetFromAddress()
+			action := args[0]
+
+			spend, err := cmd.Flags().GetBool(FlagSpend)
+			if err != nil {
+				return err
+			}
+			var msg sdk.Msg
+			if spend {
+				msg = types.NewMsgWalletSpendAction(owner, action)
+			} else {
+				msg = types.NewMsgWalletAction(owner, action)
+			}
+			err = msg.ValidateBasic()
+			if err != nil {
+				return err
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	cmd.Flags().Bool(FlagSpend, false, "Send a WalletSpendAction transaction if true")
```

Comment:
```suggestion
	cmd.Flags().Bool(FlagAllowSpend, false, "Allow the WalletAction to spend assets if specified")
```
My wording's not so great.  Alternatives in this direction would be appreciated.

Reply:
Done. Wording seems fine except trimmed the redundant "if true" at the end.